### DISCUSSION TITLE
How to link 2 tables using a FK table with supabase-js syntax?

### DISCUSSION THREAD

**Discussion by yao-dev** [supabase/supabase#11440] [Questions] [answered]
I have 3 tables

```
food
- id
- name

food_meal_type (FK table)
- id
- mealTypeId
- foodId

meal_type
- id
- name
- value
- enabled
```

in SQL I would do something similar to:

```
select * from food
inner join food_meal_type on food.id = food_meal_type."foodId"
inner join meal_type on food_meal_type."mealTypeId" = meal_type.id;
```

I tried multiple ways including the below, but I haven't found a way to achieve it

```
  const { data, error } = await supabase
    .from("food")
    .select(
      `
      id,
      name,
      mealTypes: food_meal_type(
        *,
        meal_type!mealTypeId(*)
      )
    `
    )
```

The ideal result would be

```
{
  "id": "99fc8f86-36f3-4146-b8f2-78f0f8797f95",
  "name": "Pesto fish fingers with broccoli mash",
  "mealTypes": [
    {
      "id": "a27b61c0-0367-418a-bd85-6c0b0640b89e",
      "mealTypeId": 4,
      "foodId": "99fc8f86-36f3-4146-b8f2-78f0f8797f95",
      // the fields below come from the meal_type table
      "name": "lunch",
      "value": "LUNCH"
    }
  ]
}
```

How can I do the same using the supabase-js syntax?

Thank you.

**Comment by GaryAustin1** [comment, marked as answer]
If you make mealTypeId and foodId part of the primary key in food_meal_type, PostgREST (database API) should just pick up the relationship if you use meal_type in your select.

See `Breaking change on many-to-many relationships` here: https://supabase.com/blog/postgrest-v10

Also you really should not use uppercase at all in PostgREST unless forced to by something like Prisma.

**Reply by yao-dev** [reply]
Thanks @GaryAustin1 for your help on here and on Discord as well, the solution was to set **mealTypeId** and **foodId** as PK of the **food_meal_type** table then use the syntax below with supabase-js

```
  const { data, error } = await supabase
    .from("food")
    .select(
      `
      id,
      name,
      mealTypes: meal_type(*)
    `
    )
    .limit(1)
    .single();
```

to get the following result

```
{
  "id": "04406061-f730-4145-9bc5-55f1e445a4e1",
  "name": "Warm sprout and spinach salad",
  "mealTypes": [
    {
      "id": 4,
      "name": "dinner",
      "value": "DINNER",
      "enabled": true
    }
  ]
}
```
